runtime/ast/inspect.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ast
+
+type inspector func(Element) bool
+
+func (f inspector) Walk(element Element) Walker {
+	if f(element) {
+		return f
+	}
+
+	return nil
+}
+
+func Inspect(element Element, f func(Element) bool) {
+	Walk(inspector(f), element)
+}
```

Comment:
```suggestion
func Inspect(element Element, f inspector) {
	Walk(f, element)
}
```

Reply:
The `inspector` type is private, so I think it's better to keep it as is (just like in Go's codebase)